programs/bpf_loader/src/syscalls.rs

```diff
@@ -1684,7 +1684,7 @@ fn check_authorized_program(
         || bpf_loader::check_id(program_id)
         || bpf_loader_deprecated::check_id(program_id)
         || (bpf_loader_upgradeable::check_id(program_id)
-            && !bpf_loader_upgradeable::is_upgrade_instruction(instruction_data))
+            && !bpf_loader_upgradeable::is_cpi_authorized_instruction(instruction_data))
```

Comment:
Needs to be feature gated

Reply:
Done in 05caf3b8f35ca993b31f3610d4f1902718056639 please check if that's correct